src/main/java/com/blackducksoftware/integration/hub/alert/web/controller/ConfigController.java

```diff
@@ -22,92 +22,79 @@
-            if (foundEntity != null) {
-                return Arrays.asList(databaseModelToRestModel(foundEntity));
+    public ResponseEntity<String> postConfig(final R restModel) throws IntegrationException {
+        if (!configActions.doesConfigExist(restModel.getId())) {
+            final Map<String, String> validationResults = configActions.validateConfig(restModel);
+            if (validationResults.isEmpty()) {
+                configActions.customTriggers(restModel);
+                final D updatedEntity = configActions.saveConfig(restModel);
+                return createResponse(HttpStatus.CREATED, updatedEntity.getId(), "Created");
             } else {
-                return Collections.emptyList();
+                final ResponseBodyBuilder responseBuilder = new ResponseBodyBuilder(configActions.objectTransformer.stringToLong(restModel.getId()), "There were errors with the configuration.");
+                responseBuilder.putErrors(validationResults);
+                return new ResponseEntity<>(responseBuilder.build(), HttpStatus.BAD_REQUEST);
             }
         }
-        return databaseModelsToRestModels(repository.findAll());
-    }
-
-    public ResponseEntity<String> postConfig(final R restModel) {
-        if (restModel.getId() == null || !repository.exists(restModel.getId())) {
-            ResponseEntity<String> response = validateConfig(restModel);
-            if (response == null) {
-                final D createdEntity = repository.save(restModelToDatabaseModel(restModel));
-                response = createResponse(HttpStatus.CREATED, createdEntity.getId(), "Created.");
-            }
-            return response;
-        }
         return createResponse(HttpStatus.CONFLICT, restModel.getId(), "Provided id must not be in use. To update an existing configuration, use PUT.");
     }
 
-    public ResponseEntity<String> putConfig(final R restModel) {
-        if (restModel.getId() != null && repository.exists(restModel.getId())) {
-            final D modelEntity = restModelToDatabaseModel(restModel);
-            modelEntity.setId(restModel.getId());
-            ResponseEntity<String> response = validateConfig(restModel);
-            if (response == null) {
-                final D updatedEntity = repository.save(modelEntity);
-                response = createResponse(HttpStatus.ACCEPTED, updatedEntity.getId(), "Updated.");
+    public ResponseEntity<String> putConfig(final R restModel) throws IntegrationException {
+        if (configActions.doesConfigExist(restModel.getId())) {
+            final Map<String, String> validationResults = configActions.validateConfig(restModel);
+            if (validationResults.isEmpty()) {
+                configActions.customTriggers(restModel);
+                final D updatedEntity = configActions.saveConfig(restModel);
+                return createResponse(HttpStatus.CREATED, updatedEntity.getId(), "Updated");
```

Comment:
I initially chose to use HttpStatus.CREATED here as well, but I believe you changed it to HttpStatus.ACCEPTED later, which is a more accurate response. Unless you have a reason for changing this back, I would say this is a mistake.